src/commands/source/download.rs

```diff
@@ -277,39 +297,69 @@ pub async fn download(
                     }
 
                     if source_path_exists && !force {
-                        Err(anyhow!("Source exists: {}", source.path().display()))
+                        info!("Source already exists: {}", source.path().display());
+                        Ok(DownloadResult::Skipped)
                     } else {
+                        let mut download_result: Option<DownloadResult> = None;
+
                         if source_path_exists
                         /* && force is implied by 'if' above*/
                         {
                             source.remove_file().await?;
+                            download_result = Some(DownloadResult::Forced)
                         }
 
                         progressbar.lock().await.inc_download_count().await;
                         {
                             let permit = download_sema.acquire_owned().await?;
                             perform_download(&source, progressbar.clone(), timeout).await?;
                             drop(permit);
+
+                            if download_result.is_none() {
+                                download_result = Some(DownloadResult::Succeeded)
+                            }
                         }
                         progressbar.lock().await.finish_one_download().await;
-                        Ok(())
+                        Ok(download_result.unwrap())
                     }
                 }
             })
         })
         .collect::<futures::stream::FuturesUnordered<_>>()
-        .collect::<Vec<Result<()>>>()
-        .await
-        .into_iter()
-        .collect::<Result<()>>();
+        .collect::<Vec<Result<DownloadResult>>>()
```

Comment:
Hm, this is basically `Vec<Result<Result>>` since `DownloadResult` is also a form of `Result` (with multiple `Ok`s though) - can't we simplify this to `Vec<DownloadResult>` by tracking the errors via `DownloadResult::Error(e)`?
Flattening the data structure seems nicer but there might be valid reasons for aborting early via `Result`.